cmd/observer/node_builder/observer_builder.go

```diff
@@ -0,0 +1,1043 @@
+
+			msgValidators := publicNetworkMsgValidators(node.Logger, node.IdentityProvider, observerNodeID)
+
+			builder.initMiddleware(observerNodeID, node.Metrics.Network, libP2PFactory, msgValidators...)
+
+			return nil
+		})
+}
+
+// Build enqueues the sync engine and the follower engine for the observer.
+// Currently, the observer only runs the follower engine.
+func (builder *ObserverServiceBuilder) Build() (cmd.Node, error) {
+	builder.BuildConsensusFollower()
+	if builder.executionDataSyncEnabled {
+		builder.BuildExecutionDataRequester()
+	}
+	return builder.FlowNodeBuilder.Build()
+}
+
+// enqueuePublicNetworkInit enqueues the observer network component initialized for the observer
+func (builder *ObserverServiceBuilder) enqueuePublicNetworkInit() {
+
+	builder.Component("unstaked network", func(node *cmd.NodeConfig) (module.ReadyDoneAware, error) {
+		var heroCacheCollector module.HeroCacheMetrics = metrics.NewNoopCollector()
+		if builder.HeroCacheMetricsEnable {
+			heroCacheCollector = metrics.NetworkReceiveCacheMetricsFactory(builder.MetricsRegisterer)
+		}
+		receiveCache := netcache.NewHeroReceiveCache(builder.NetworkReceivedMessageCacheSize,
+			builder.Logger,
+			heroCacheCollector)
+
+		err := node.Metrics.Mempool.Register(metrics.ResourceNetworkingReceiveCache, receiveCache.Size)
+		if err != nil {
+			return nil, fmt.Errorf("could not register networking receive cache metric: %w", err)
+		}
+
+		// topology is nil since it is automatically managed by libp2p
+		net, err := builder.initNetwork(builder.Me, builder.Metrics.Network, builder.Middleware, nil, receiveCache)
+		if err != nil {
+			return nil, err
+		}
+
+		builder.Network = converter.NewNetwork(net, engine.SyncCommittee, engine.PublicSyncCommittee)
+
+		builder.Logger.Info().Msgf("network will run on address: %s", builder.BindAddr)
+
+		idEvents := gadgets.NewIdentityDeltas(builder.Middleware.UpdateNodeAddresses)
+		builder.ProtocolEvents.AddConsumer(idEvents)
+
+		return builder.Network, nil
+	})
+}
+
+// enqueueConnectWithStakedAN enqueues the upstream connector component which connects the libp2p host of the observer
+// service with the AN.
+// Currently, there is an issue with LibP2P stopping advertisements of subscribed topics if no peers are connected
+// (https://github.com/libp2p/go-libp2p-pubsub/issues/442). This means that an observer could end up not being
+// discovered by other observers if it subscribes to a topic before connecting to the AN. Hence, the need
+// of an explicit connect to the AN before the node attempts to subscribe to topics.
+func (builder *ObserverServiceBuilder) enqueueConnectWithStakedAN() {
+	builder.Component("upstream connector", func(_ *cmd.NodeConfig) (module.ReadyDoneAware, error) {
+		return consensus_follower.NewUpstreamConnector(builder.bootstrapIdentities, builder.LibP2PNode, builder.Logger), nil
+	})
+}
+
+func (builder *ObserverServiceBuilder) attachRPCEngine() {
```

Comment:
nit: be consistent with other method names
```suggestion
func (builder *ObserverServiceBuilder) enqueueRPCServer() {
```

Reply:
Addressed: https://github.com/GetElastech/flow-go/commit/089c646d408f9d691621ca825255238e51745fbc

Reply:
It also addressed the node id log indexing issue above [:7] [:8]